lib/tty/pager/basic.rb

```diff
@@ -43,50 +45,71 @@ def default_prompt
       #
       # @api public
       def page(text, &callback)
-        page_num = 1
-        leftover = []
-        lines_left = @height
+        write(text, &callback)
+        reset
+      end
 
+      def write(text, &callback)
         text.lines.each do |line|
           chunk = []
-          if !leftover.empty?
-            chunk = leftover
-            leftover = []
+          unless @state.leftover.empty?
+            chunk = @state.leftover
+            @state.leftover = []
           end
           wrapped_line = Strings.wrap(line, @width)
           wrapped_line.lines.each do |line_part|
-            if lines_left > 0
+            if @state.lines_left.positive?
```

Comment:
I know this is pain, but I'd keep the `> 0` comparison. Why? At the moment this gem is meant to work with Ruby `>= 2.0`. Sorry!

Reply:
Yeah, I'm happy to use it as well. I think I'll just go ahead and ignore future Hound complaints :).